### PR TITLE
fixes #141 - Configured tslint to lint on command and in unit tests.

### LINKED ISSUE #141: Make TS linting better
Currently the TS linting is not configured and prints out a lot of warning when you run the unit tests.

### PR DESCRIPTION
**Pull request by joshuawilson**
Add codelizer support to lint Angular 2. Configured tslint to lint on command and in unit tests.

This uses to sets of rules. The main tslint rules can be found at http://palantir.github.io/tslint/rules/

The codelizer is at https://github.com/mgechev/codelyzer. 

The fixes will be in another PR.

### CODE REVIEW

File: config/webpack.test.js [hunk 1]
```diff
@@ -204,7 +218,7 @@ module.exports = {
   tslint: {
     emitErrors: false,
     failOnHint: false,
-    resourcePath: 'src'
+    resourcePath: root
```
Comment (sanbornsen) [inline config/webpack.test.js]: I don't know what does root mean. Is it a keyword, which points to the root directory of the project? If it is so, then why are you linting over node_modules and stuff? Shouldn't we only run linting on the code we write?

File: package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "copy-files": "cpy src/index.html src/favicon.ico dist",
     "docker:run": "webpack-dev-server --inline --progress --port 8080",
     "functests": "npm-run-all -p webdriver-start testserver-start protractor",
-    "lint": "tslint ./src/*",
+    "lint": "tslint -c tslint.json --force src/**/*.ts src/**/**/*.ts src/**/**/**/*.ts src/**/**/**/**/*.ts src/**/**/**/**/**/*.ts",
```
Comment (sanbornsen) [inline package.json]: I see what are you trying to do here, but I believe just **src/**/**.ts** should do the trick.

File: config/webpack.test.js (hunk 1 above)
Comment (sanbornsen) [inline config/webpack.test.js]: 'root' is deprecated, use 'global'

**Comment by michaelkleinhenz** [comment]
Looks good to me. Maybe we have to tweak some rules, but I think we should try it with this set and see if anyone goes rampage (indenting :-)

**Comment by nimishamukherjee** [comment]
Looks good to me. Do we need linting on CSS?

File: package.json (hunk 2 above)
Comment (sanbornsen) [inline package.json]: Cool :)

**Pull request merged**